src/builder/cfg.rs

```diff
@@ -248,18 +242,9 @@ impl<B: AsMut<Hugr> + AsRef<Hugr>> BlockBuilder<B> {
     ) -> Result<(), BuildError> {
         Dataflow::set_outputs(self, [branch_wire].into_iter().chain(outputs))
     }
-    fn create(
-        base: B,
-        block_n: Node,
-        tuple_sum_rows: impl IntoIterator<Item = TypeRow>,
-        other_outputs: TypeRow,
-        inputs: TypeRow,
-    ) -> Result<Self, BuildError> {
-        // The node outputs a TupleSum before the data outputs of the block node
-        let tuple_sum_type = Type::new_tuple_sum(tuple_sum_rows);
-        let mut node_outputs = vec![tuple_sum_type];
-        node_outputs.extend_from_slice(&other_outputs);
-        let signature = FunctionType::new(inputs, TypeRow::from(node_outputs));
+    fn create(base: B, block_n: Node) -> Result<Self, BuildError> {
+        let block_op = base.get_optype(block_n).as_basic_block().unwrap();
+        let signature = block_op.inner_signature();
```

Comment:
I was a bit suspicious how this change fitted in at first but it's a nice result!